Automattic/wp-calypso
# Stats: Refactoring summary page header

Related to #72175

## Proposed Changes

* refactoring the summary page header and moving it from <StatsModule /> to the summary page component (to avoid controlling its visibility with the `summary` property and keeping the header with the page where it belongs)
* removing borders from the period select dropdown to match the new design
* when period navigation buttons don't fit the line, move it above the title instead of below (this creates a nice flow of a selection, header and data)

## Testing Instructions



* use a live branch and check a summary header across all breakpoints
* to go to a summary page, click on `View all` on a Stats module on the `Traffic` page
* check that a different type of header with a button (e.g. Video or Authors components) looks fine, keeping the button on the right side

Before|After
-|-
![SCR-20230303-ukh](https://user-images.githubusercontent.com/112354940/222684427-ff6e3dfa-4cf3-48f5-8183-c68a1d3d1412.png)|![SCR-20230303-ujr](https://user-images.githubusercontent.com/112354940/222684563-e4909dcd-7204-4f25-988b-bec99dfb20eb.png)
![SCR-20230303-uk6](https://user-images.githubusercontent.com/112354940/222684466-46c5a95f-c711-4794-bcc0-400c5dce7697.png)|![SCR-20230303-ujh](https://user-images.githubusercontent.com/112354940/222684600-d3c9eb12-8de3-4701-870f-52523e1955c1.png)
-|![SCR-20230303-uj6](https://user-images.githubusercontent.com/112354940/222684622-eb909940-5312-439a-9fff-e67856805ea0.png)
-|![SCR-20230303-uiu](https://user-images.githubusercontent.com/112354940/222684657-6e780f4c-8314-4b28-825d-0e59738c372d.png)

## Pre-merge Checklist
